src/renderer/services/price-api/coin-cap.js

```diff
@@ -0,0 +1,213 @@
+import got from 'got'
+import dayjs from 'dayjs'
+import moment from 'moment'
+import { min, max } from 'lodash'
+import { convert } from 'cashify'
+import { MARKET } from '@config'
+
+// All prices on the CoinCap API are standardized in USD (United States Dollar)
+const BASE_CURRENCY = 'USD'
+const AMOUNT_TO_CONVERT = 1
+
+export default class CoinCapAdapter {
+  /**
+   * Get token name from API
+   * @param  {String} token
+   * @return {(String|null)}
+   */
+  static async getTokenId (token, limit = 1000) {
+    if (this.tokenLookup) {
+      return this.tokenLookup[token.toUpperCase()]
+    }
+
+    try {
+      const uri = `${MARKET.source.coinCap}/assets?limit=${limit}`
+      const { body } = await got(uri, {
+        json: true
+      })
+      const { data } = body
+
+      this.tokenLookup = data.reduce((map, value, index) => {
+        map[value.symbol.toUpperCase()] = value.id
+
+        return map
+      }, {})
+
+      return this.tokenLookup[token.toUpperCase()]
+    } catch (error) {
+      return null
+    }
+  }
+
+  /**
+   * Get Ark data from API
+   * @return {(Object|null)}
+   */
+  static async fetchArkData () {
+    try {
+      const uri = `${MARKET.source.coinCap}/assets/ark`
+      const { body } = await got(uri, {
+        json: true
+      })
+      const { data } = body
+
+      return data
+    } catch (error) {
+      return null
+    }
+  }
+
+  /**
+   * Get assets and rates data from API
+   * @return {(Object|null)}
+   */
+  static async getCurrencyData () {
+    try {
+      const uri = `${MARKET.source.coinCap}/rates`
+      const { body } = await got(uri, {
+        json: true
+      })
+      const { data, timestamp } = body
+      const arkData = await this.fetchArkData()
+
+      const assets = data.reduce((map, value, index) => {
```

Comment:
Instead of 2 reduce loops, do 1 (removing the hardcoded ark references):

```
  const response = {
    assets: { [arkData.symbol.toUpperCase()]: arkData },
    rates: { [arkData.symbol.toUpperCase()]: arkData.rateUsd },
    timestamp
  }

  foreach (const value of data) {
    response.assets[value.symbol.toUpperCase()] = value
    response.rates[value.symbol.toUpperCase()] = value.rateUsd
  }

  return response
```